lib/liquid/block.rb

```diff
@@ -40,11 +59,27 @@ def parse(tokens)
             raise SyntaxError, "Tag '#{token}' was not properly terminated with regexp: #{TagEnd.inspect} "
           end
         when IsVariable
+          @@whitespace = false
+          if token =~ IsWhiteSpaceVariableEnd
+            @@whitespace = true
+          end
+          if token =~ IsWhiteSpaceVariableStart
+            previous_token = @nodelist.pop
+            if previous_token.is_a?(String)
+              previous_token.gsub!(WhiteSpaceEnd, '')
+              @nodelist << previous_token
+            end
```

Comment:
This is doing the same thing as for tags, so the implementation should be shared.  Even the check for the control character could be shared since we we just need to check the third byte for a `-` in both cases.